### ISSUE TITLE
echo 0;

### ISSUE THREAD

**Issue by vstepanyuk** [bobthecow/psysh#289]
```
>>> echo 0
=> null
```

```
>>> echo 1
1⏎
=> null
```

**Comment by vstepanyuk** [comment]
```
>>> echo 1
1⏎
=> null
```

**Comment by bobthecow** [comment]
Hmm. Interesting. I'll take a look.

**Comment by GrahamCampbell** [comment]
This is NOT a bug in psysh. This is just how php's echo function works. 0 is cast to empty string, and 1 is cast to the string "1"/

**Comment by vstepanyuk** [comment]
```
~ ❯❯❯ php -a
Interactive shell

php > echo 0;
0
php >
```

**Comment by GrahamCampbell** [comment]
Hmmm. You're right. Very strange.

**Comment by bobthecow** [comment]
Not all that strange. PHP's stupid `empty` implementation strikes again.

https://github.com/bobthecow/psysh/blob/master/src/Psy/Shell.php#L605

**Comment by vstepanyuk** [comment]
http://php.net/empty

"" (an empty string)
0 (0 as an integer)
0.0 (0 as a float)
"0" (0 as a string)
NULL
FALSE
array() (an empty array)
$var; (a variable declared, but without a value)

**Comment by bobthecow** [comment]
Oh, I know why it does this. It's still stupid :)

**Comment by bobthecow** [comment]
Each of those except `'0'` is an empty version of whatever type they are. Empty string? `''`. Empty array? `[]`.

`'0'` is _not_ an empty string.

**Comment by vstepanyuk** [comment]
it's also:)
```
php > var_dump(empty('0'));
bool(true)
php > var_dump(empty('\0'));
bool(false)
```

**Comment by vstepanyuk** [comment]
ah.. understand what you mean

**Comment by vstepanyuk** [comment]
but I think php tries to cast to int, I don't understand why

**Comment by bobthecow** [comment]
Because PHP made bad decisions a long time ago, and never breaks backwards compatibility.

**Comment by vstepanyuk** [comment]
lol. true)

**Comment by bobthecow** [comment]
the `'\0'` one is interesting. to me, that's more empty than `'0'` :P

**Comment by vstepanyuk** [comment]
exactly

**Comment by bobthecow** [comment]
OK. Should be fixed in `develop`. The only thing I'm not sure about is the `===` on the string comparison… I couldn't get output buffering to send the `ob_start` callback anything but a string, but i'm not positive it'd never get `null` or `false` something dumb. Could add an explicit `(string)` thunk just in case.

**Comment by vstepanyuk** [comment]
Thx. Nice repl anyway

**Comment by bobthecow** [comment]
The fix was released in v0.7.1. Thanks again for reporting!
